optimum/amd/ryzenai/modeling.py

```diff
@@ -623,6 +624,48 @@ def forward(self, pixel_values):
 
         return ImageClassifierOutput(logits=next(iter(outputs.values())))
 
+    @classmethod
+    def _export(
+        cls,
+        model_id: str,
+        config: "PretrainedConfig" = None,
+        use_auth_token: Optional[Union[bool, str]] = None,
+        revision: Optional[str] = None,
+        force_download: bool = False,
+        cache_dir: Optional[str] = None,
+        subfolder: str = "",
+        # use_auth_token=use_auth_token,
+        local_files_only: bool = False,
+        trust_remote_code: bool = False,
+        vaip_config: Optional[str] = None,
+        # vaip_config=vaip_config,
+        provider: str = "CPUExecutionProvider",
+        session_options: Optional[ort.SessionOptions] = None,
+        provider_options: Optional[Dict[str, Any]] = None,
+        use_io_binding: Optional[bool] = None,
+        task: Optional[str] = None,
+    ) -> "RyzenAIModel":
+
+        save_dir = TemporaryDirectory()
+        save_dir_path = Path(save_dir.name)
+        main_export(
+            model_name_or_path=model_id,
+            output=save_dir_path,
+            task="image-classification",
+            opset=17,
+            batch_size=1,
+            no_dynamic_axes=True,
+        )
+        config = PretrainedConfig.from_pretrained(save_dir_path)
+        return cls._from_pretrained(
+            save_dir_path,
+            config,
+            model_save_dir=save_dir,
+            provider="CPUExecutionProvider",
+            session_options=session_options,
+            provider_options=provider_options,
+        )
```

Comment:
Could you ensure the signature for `main_export` and `cls._from_pretrained` matches the code here : https://github.com/huggingface/optimum/blob/main/optimum/onnxruntime/modeling_ort.py#L562

and 

https://github.com/huggingface/optimum-amd/blob/main/optimum/amd/ryzenai/modeling.py#L498 (This could contain the additional option for `vaip_config`)

In case the model is given from hub the other arguments are necessary.

Reply:
Thanks for the help. I've made the changes